src/content/docs/en/guides/troubleshooting.mdx

```diff
@@ -132,6 +132,22 @@ Yarn 2+, a.k.a. Berry, uses a technique called [Plug'n'Play (PnP)](https://yarnp
 nodeLinker: "node-modules"
 ```
 
+### Using Astro with dependencies in the root of a Monorepo
+
+When installing dependencies in a Monorepo, it is possible to also install dependencies in the root `package.json` file instead of the project `package.json`: installing dependencies this way can make some Astro packages to not work or bundle as expected.
+It is recommended to just put all of Astro packages in the project `package.json`to avoid this kind of issue; if it can't be avoided you will probably need to manually put the dependencies in the `vite.ssr.noExternal` part of Astro's config:
+```js
+// astro.config.mjs
+import { defineConfig } from 'astro/config'
+
+export default defineConfig({
+  vite: {
+    ssr: {
+      noExternal: ['@astrojs/package-a', '@astrojs/package-b', ..., '@astrojs/package-N']
+})
+```
+
+
```

Comment:
```suggestion
### Using Astro in a monorepo

When working with Astro in a monorepo setup, it is recommended to keep all the dependencies in the project `package.json` file.
However, you may also want to use Astro in the root of the monorepo (for example, to generate static routes properly). In this case, you may encounter some issues with dependencies not being installed and bundled properly.
To avoid this, you need to manually add these dependencies to the `vite.ssr.noExternal` part of Astro's config:

```js
// astro.config.mjs
import { defineConfig } from 'astro/config'
export default defineConfig({
  vite: {
    ssr: {
      noExternal: [
        '@astrojs/package-a',
        '@astrojs/package-b',
        '@astrojs/package-n',
      ]
    }
  }
})
```

```

Reply:
Thanks for this helpful suggestion, @bandantonio ! I'm going to commit it to make sure you are on record as a co-author, and then do some smaller edits myself.

Team work!  :partying_face: